apps/web/src/pages/templates/workflow/digest/WillBeSentHeader.tsx

```diff
@@ -0,0 +1,175 @@
+import { useCallback, useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import * as capitalize from 'lodash.capitalize';
+import { useMantineColorScheme } from '@mantine/core';
+import { DigestTypeEnum, DigestUnitEnum, MonthlyTypeEnum } from '@novu/shared';
+
+import { colors } from '../../../../design-system';
+import { pluralizeTime } from '../../../../utils';
+
+const getOrdinalValueLabel = (value: string) => {
+  if (value === 'day' || value === 'weekday') {
+    return value;
+  }
+  if (value === 'weekend') {
+    return 'weekend day';
+  }
+
+  return capitalize(value);
+};
+
+const getOrdinal = (num: string | number) => {
+  if (typeof num === 'string') {
+    const res = parseInt(num, 10);
+
+    if (isNaN(res)) {
+      return num;
+    }
+    num = res;
+  }
+  const ord = ['st', 'nd', 'rd'];
+  const exceptions = [11, 12, 13];
+  let nth = ord[(num % 10) - 1];
+
+  if (ord[(num % 10) - 1] == undefined || exceptions.includes(num % 100)) {
+    nth = 'th';
+  }
+
+  return num + nth;
+};
+
+const Highlight = ({ children }) => {
+  const { colorScheme } = useMantineColorScheme();
+
+  return (
+    <b
+      style={{
+        color: colorScheme === 'dark' ? colors.B80 : colors.B40,
+      }}
+    >
+      {children}
+    </b>
+  );
+};
+export const WillBeSentHeader = ({ index }) => {
+  const { watch, setValue } = useFormContext();
+
+  const type = watch(`steps.${index}.metadata.type`);
+  const unit = watch(`steps.${index}.metadata.unit`);
+  const backoff = watch(`steps.${index}.metadata.backoff`);
+  const atTime = watch(`steps.${index}.metadata.timed.atTime`);
+  const weekDays = watch(`steps.${index}.metadata.timed.weekDays`) || [];
+  const monthDays = watch(`steps.${index}.metadata.timed.monthDays`) || [];
+  const amount = watch(`steps.${index}.metadata.amount`);
+
+  useEffect(() => {
+    if (!backoff) {
+      return;
+    }
+    if (type !== DigestTypeEnum.TIMED) {
+      return;
+    }
+    setValue(`steps.${index}.metadata.backoff`, false);
+  }, [backoff, type, index]);
+
+  const BackoffText = useCallback(() => {
+    return backoff ? (
+      <>
+        only if <Highlight>frequent events</Highlight> occur
+      </>
+    ) : null;
+  }, [backoff]);
+
+  if (type === DigestTypeEnum.TIMED && unit == DigestUnitEnum.HOURS) {
```

Comment:
Seeing all the conditions applied in this component seems to meant to be only used in the Timed Digest case. Should be better to remove all the checks of the digest type and do the control of the type of digest in any parent component that uses this?